securecookie.go

```diff
@@ -421,7 +347,9 @@ func decrypt(block cipher.Block, value []byte) ([]byte, error) {
 
 // Serialize encodes a value using gob.
 func (e GobEncoder) Serialize(src interface{}) ([]byte, error) {
-	buf := new(bytes.Buffer)
+	buf := getBuffer()
+	defer bufferPool.Put(buf)
```

Comment:
Be aware that the `buf.Bytes()` further down returns the backing storage of the bytes.Buffer. If the buffer in the pool gets reused, the user will read garbage from the returned `[]byte`.